functorch/csrc/BatchRulesBinaryOps.cpp

```diff
@@ -239,6 +239,31 @@ std::tuple<Tensor,optional<int64_t>> masked_select_batch_rule(
   return std::make_tuple(result, 0);
 }
 
+std::tuple<Tensor,optional<int64_t>> masked_select_backward_batch_rule(
+    const Tensor& grad, optional<int64_t> grad_bdim,
+    const Tensor& self, optional<int64_t> self_bdim,
+    const Tensor& mask, optional<int64_t> mask_bdim) {
+  TORCH_CHECK(!mask_bdim.has_value(),
+      "vmap: Attempted to vmap over `mask` in torch.masked_select_backward(grad, self, mask) ",
+      "We cannot support this because for each batch this would return a ",
+      "differently shaped Tensor. "
+      "Please voice your support in https://github.com/pytorch/functorch/issues/256");
+  auto self_ = moveBatchDimToFront(self, self_bdim);
+  auto grad_ = moveBatchDimToFront(grad, grad_bdim);
+
+  const auto self_logical_rank = rankWithoutBatchDim(self, self_bdim);
+  const auto max_logical_rank = std::max(self_logical_rank, mask.dim());
+
+  self_ = maybePadToLogicalRank(self_, self_bdim, max_logical_rank);
+
+  const auto batch_size = get_bdim_size2(grad, grad_bdim, self, self_bdim);
+  self_ = ensure_has_bdim(self_, self_bdim.has_value(), batch_size);
+  grad_ = ensure_has_bdim(grad_, grad_bdim.has_value(), batch_size);
+
+  const auto result = at::masked_select_backward(grad_, self_.contiguous(), mask);
```

Comment:
It was putting the result in the wrong shape without the `.contiguous`. Seems like a bug?

- [ ] file issue

Reply:
Classic pytorch...